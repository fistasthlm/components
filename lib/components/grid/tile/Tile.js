import React from 'react'
import ImmutablePropTypes from 'react-immutable-proptypes'
import { Link } from 'react-router-dom'
import InstagramLink from '../../shared/instagramLink/InstagramLink'
import Image from '../../shared/image/Image'
import style from './Tile.scss'

export default function Tile ({bike}) {
  const hasImage = bike.get('images').first()

  return (
    <div className={style.tile}>
      <Link to={`/bike/${bike.get('_id')}`}>
        <div className={style.image}>
          {
            hasImage &&
            <Image
              url={hasImage.get('url')}
              className={style.picture}
              resize={true}
              width="500"
              caption={bike.get('title')} />
          }

        </div>
        <h2 className={style.title}>
          {bike.get('title')}
        </h2>
      </Link>
      <InstagramLink handle={bike.get('instagram')} />
    </div>
  )
}

Tile.propTypes = {
  bike: ImmutablePropTypes.map.isRequired
}
