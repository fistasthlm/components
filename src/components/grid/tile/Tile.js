import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import { Map } from 'immutable'
import InstagramLink from 'components/shared/instagramLink/InstagramLink'
import Image from 'components/shared/image/Image'
import style from './Tile.scss'

export default function BikeTile ({bike}) {
  const hasImage = bike.get('images').first()

  return (
    <div className={style.tile}>
      <Link to={`/bike/${bike.get('_id')}`}>
        <div className={style.image}>
          {
            hasImage &&
            <Image
              url={hasImage.get('url')}
              className={style.picture}
              resize={true}
              width="500"
              caption={bike.get('title')} />
          }

        </div>
        <h2 className={bike.title}>
          {bike.get('title')}
        </h2>
      </Link>
      <InstagramLink handle={bike.get('instagram')} />
    </div>
  )
}

BikeTile.propTypes = {
  bike: PropTypes.instanceOf(Map).isRequired
}
