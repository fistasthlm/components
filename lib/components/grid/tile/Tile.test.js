import React from 'react'
import { shallow } from 'enzyme'
import { Map, List } from 'immutable'
import Tile from './Tile'

const setup = propOverrides => {
  const props = Object.assign({
    bike: Map({
      images: List(),
      instagram: 'pete',
    }),
  }, propOverrides)

  return shallow(<Tile {...props} />)
}

describe('Render', () => {
  it('should render component', () => {
    const wrapper = setup()

    expect(wrapper).toMatchSnapshot()
  })
})
