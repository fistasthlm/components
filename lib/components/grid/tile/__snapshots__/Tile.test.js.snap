// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<div>
  <Link
    replace={false}
    to="/bike/undefined"
  >
    <div />
    <h2 />
  </Link>
  <InstagramLink
    handle="pete"
  />
</div>
`;
